tck/features/expressions/graph/Graph9.feature

```diff
@@ -28,94 +28,84 @@
 
 #encoding: utf-8
 
-Feature: Graph9 - Property existence check
+Feature: Graph9 - Retrieve all properties as a property map
 
-  Scenario: [1] `exists()` with dynamic property lookup
+  Scenario: [1] `properties()` on a node
     Given an empty graph
     And having executed:
       """
-      CREATE (:Person {name: 'foo'}),
-             (:Person)
+      CREATE (n:Person {name: 'Popeye', level: 9001})
       """
     When executing query:
       """
-      MATCH (n:Person)
-      WHERE exists(n['name'])
-      RETURN n
+      MATCH (p:Person)
+      RETURN properties(p) AS m
       """
     Then the result should be, in any order:
-      | n                       |
-      | (:Person {name: 'foo'}) |
+      | m                             |
+      | {name: 'Popeye', level: 9001} |
     And no side effects
 
-  Scenario: [2] `exists()` is case insensitive
+  Scenario: [2] `properties()` on a relationship
     Given an empty graph
     And having executed:
       """
-      CREATE (a:X {prop: 42}), (:X)
+      CREATE (n)-[:R {name: 'Popeye', level: 9001}]->(n)
       """
     When executing query:
       """
-      MATCH (n:X)
-      RETURN n, EXIsTS(n.prop) AS b
+      MATCH ()-[r:R]->()
+      RETURN properties(r) AS m
       """
     Then the result should be, in any order:
-      | n               | b     |
-      | (:X {prop: 42}) | true  |
-      | (:X)            | false |
+      | m                             |
+      | {name: 'Popeye', level: 9001} |
     And no side effects
 
-  Scenario: [3] Property existence check on non-null node
-    Given an empty graph
-    And having executed:
-      """
-      CREATE ({exists: 42})
-      """
+  Scenario: [3] `properties()` on null
+    Given any graph
```

Comment:
This is too loose. There are conceivably graphs where `(:DoesNotExist)` nodes exist.
```suggestion
    Given an empty graph
```

Reply:
Right!